src/checks.ml

```diff
@@ -0,0 +1,203 @@
+include Checks_intf
+
+module Make (K : Data.Key) (V : Data.Value) (IO : Io.S) = struct
+  module Entry = Data.Entry.Make (K) (V)
+
+  module IO = struct
+    include Io.Extend (IO)
+
+    (** This module never makes persistent changes *)
+    let v =
+      v ~fresh:false ~readonly:true ?flush_callback:None ~generation:0L
+        ~fan_size:0L
+
+    let iter ?min ?max f =
+      iter ?min ?max
+        ~page_size:Int64.(mul (of_int Entry.encoded_size) 1000L)
```

Comment:
To make this similar to the [`iter` in `index.ml`](https://github.com/mirage/index/pull/236/files#diff-488de9f5720c03d7021bc9e6e186bd0f516269ec43fe0f4cea0e82e7bd495063R79) I would define a separate `page_size` and have `f off entry`.

Reply:
Yep, thanks.